Add 3d-API-Doc workflow into engine

ColinCollins opened the pull request:
先不要合并，这是一个讨论式 pr！

重点查看 gfx.ts 和 creator.d.ts 两个文件中关于 module 的设置。（先不要在意 gfx 里面的内容，我为了方面查看效果构建的测试 .ts）
![image](https://user-images.githubusercontent.com/35832931/53095091-e1026180-3556-11e9-9fd0-12510d2fbdb8.png)

![image](https://user-images.githubusercontent.com/35832931/53095326-7e5d9580-3557-11e9-97b2-47494a53e21d.png)

![image](https://user-images.githubusercontent.com/35832931/53095340-87e6fd80-3557-11e9-8d92-d19865cfdcb7.png)


1. 3D 文档构建的话建议是在同一 module ts **文件群** 内定义 export module name
2. 之后我通过 creator.d.ts 中定义 cc.subModuleName 来实现分模块

ColinCollins committed:
Add 3d-API-Doc workflow into engine

jareguo commented on package.json:
```diff
@@ -35,8 +36,10 @@
     "async": "1.5.2",
     "babel-preset-env": "^1.2.2",
     "babelify": "7.3.0",
+    "browserify": "^16.2.3",
```
我没看到你在哪用到  browserify

ColinCollins replied:
这个 pr 不用看了，不会合并的，正式的提交我会放在另一个 pr 里面

ColinCollins replied:
已经打算重新修改。。。稍后修正多余的部分

jareguo commented on package.json:
```diff
@@ -35,8 +36,10 @@
+    "browserify": "^16.2.3",
     "cannon": "^0.6.2",
     "chalk": "1.1.0",
+    "typedoc": "github:ColinCollins/typedoc#master",
```
在 Creator 官方帐号下建一个仓库，不要用个人账户。你有建官方仓库的权限

jareguo commented on package.json:
```diff
@@ -70,6 +74,10 @@
     "tslint": "^5.12.0",
+    "typedoc": "^0.14.2",
+    "typedoc-plugin-external-module-name": "^2.0.0",
+    "typedoc-plugin-internal-external": "^2.0.1",
```
我没看到你在 typedoc.json 用到以上两个插件

ColinCollins replied:
以后会用到的。。。。typedoc 会自动的扫描当前 node_modules 存在的 typedoc-plugins

jareguo commented on typedoc.json:
```diff
@@ -0,0 +1,9 @@
+{
+    "mode": "modules",
+    "out": "docs-3d",
+	  "ignoreCompilerErrors": true,
```
注意缩进，太难看了

ColinCollins committed:
Update the modify

1. increase the api docs generator
2. please check the cocos-creator/typedoc repository,  make sure the modify has been merge
